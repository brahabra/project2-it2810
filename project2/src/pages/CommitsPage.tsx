import { useEffect, useState } from "react";
import "../styles/CommitsPage.css";
import { CommitComponent } from "../components/CommitComponent";
import { Graph } from "../components/Graph";
import DateRangePicker from "../components/DateRangePicker";
import { getCommits } from "../api/fetch";
import { Commit } from "../types";
import { Box } from "@mui/material";
import { Dayjs } from "dayjs";
import NameRowComponent from "../components/NameRowComponent";

export default function CommitsPage() {
  const [commits, setCommits] = useState<Commit[]>([]);
  const [isLoading, setLoading] = useState(true);
  const [filterName, setName] = useState("");
  const [filterList, setFilterList] = useState<Commit[]>([]);
  const [startValue, setStartValue] = useState<Dayjs | null>(null);
  const [endValue, setEndValue] = useState<Dayjs | null>(null);

  const ComponentList = () => {
    return (
      <Box sx={{ width: "100%", margin: "auto", backgroundColor: "#DAF7A6" }}>
        {filterList.map((commit) => (
          <CommitComponent key={commit.id} commit={commit} />
        ))}
      </Box>
    );
  };

  function filterCommitList() {
    const commitsCopy = [...commits];
    let filteredCommits: Commit[] = commitsCopy;
    if (filterName && filterName != "default") {
      filteredCommits = filteredCommits.filter(
        (commit) => commit.author_name == filterName
      );
    }

    if (startValue) {
      filteredCommits = filteredCommits.filter(
        (commit) =>
          startValue.isBefore(commit.created_at) ||
          startValue.isSame(commit.created_at, "day")
      );
    }

    if (endValue) {
      filteredCommits = filteredCommits.filter(
        (commit) =>
          endValue.isAfter(commit.created_at) ||
          endValue.isSame(commit.created_at, "day")
      );
    }
    setFilterList(filteredCommits);
  }

  useEffect(() => {
    getCommits("17381", "glpat-CRs4epaLyzKdvdpGzE_3", "main").then(
      (res: Commit[]) => {
        setCommits(res);
        setLoading(false);
      }
    );
  }, []);

  useEffect(() => {
    filterCommitList();
  }, [commits, filterName, startValue, endValue]);

  return (
    <div>
      <div className="header">
        <h2>Commits</h2>
      </div>
      <div className="dateRange">
        <DateRangePicker
          startValue={startValue}
          setStartValue={setStartValue}
          endValue={endValue}
          setEndValue={setEndValue}
        />
      </div>
      {isLoading ? null : (
        <NameRowComponent
          filterName={filterName}
          commits={commits}
          setName={setName}
        />
      )}
      <div className="commits">
        {isLoading ? <p>Loading data ...</p> : <ComponentList />}
      </div>
    </div>
  );
}
