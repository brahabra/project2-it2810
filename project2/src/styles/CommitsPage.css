.header {
    text-align: center;
}

.commitContainer {
    margin: auto;
    display: flex;
    gap: 15px;
    justify-content: space-evenly;
    flex-direction: row;
    width: 75vw;

}

@media only screen and (max-width: 850px) {
    .commitContainer {
        flex-direction: column;

    }
  }