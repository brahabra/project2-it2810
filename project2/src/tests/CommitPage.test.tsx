import * as React from 'react'
import CommitsPage from '../pages/CommitsPage'

const ReactTestRenderer = require('react-test-renderer');

it('renders correctly', () => {
  const tree = ReactTestRenderer.create(<CommitsPage/>).toJSON()
  expect(tree).toMatchSnapshot()
})