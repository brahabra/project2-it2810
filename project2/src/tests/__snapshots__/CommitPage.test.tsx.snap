// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="header"
  >
    <h2>
      Commits
    </h2>
    <p>
      Showing the 100 last commits in your repository to main branch
    </p>
  </div>
  <p>
    Loading ...
  </p>
</div>
`;
